### ISSUE TITLE
Reader not correctly reading in row

### ISSUE THREAD

**Issue by mherbold** [thephpleague/csv#433]
Using 9.7.1

Trying to read in the following row - 

record,,2,1728,2,"Meeting with Jenn and Patty re: financials",,Yes,0,0,0,0,0,"Does Not Matter",2157,"\\",No,6775,560,"Meeting with Jenn and Patty",1,No,No,No,0,"Jenn P office",Confirmed,America/New_York,4,,10,,No,No,0,50,,No,No,,0,,,No

Excel reads in that row just fine.  League messes up at the "\\" - it thinks that the quote is being escaped, when it is not.  How do I fix this - or how do I tell League to read this in properly? 

RFC 4180 says about escaping characters - If double-quotes are used to enclose fields, then a double-quote appearing inside a field must be escaped by preceding it with another double quote.

In other words, backslashes are never used to escape double quote characters with RFC 4180 CSV files.  How do I turn backslash escaping of double quotes off on the reader?

**Comment by nyamsprod** [comment]
@mherbold thanks for using the library did you look into the documentation the page about the escape character usage ? Check this URI: https://csv.thephpleague.com/9.0/connections/controls/#the-escape-character

**Comment by mherbold** [comment]
@nyamsprod Hello - yes I saw that and I tried setting the escape character to be "" to turn it off, however that breaks the reading in of some other rows, especially the ones that contain binary data in cells.  I am not sure how to fix this.

The CSV file is being generated by an older version of this library because that site is running on PHP 5.6.  It's running version 8.2.3.  Perhaps the error is on the writing side of things?  Do you remember what it could be?

**Comment by nyamsprod** [comment]
@mherbold I think you are running into the famous escape bug because that escape character was wrongly added to PHP csv parsing. The other work around would be to use [some stream filtering](https://csv.thephpleague.com/8.0/filtering/) but it's been ages since I even touch that part for CSV 8.0 maybe use [the same functionality  as exposed in v9](https://csv.thephpleague.com/9.0/connections/filters/).

**Comment by PascalBrouwers** [comment]
Running into the exact same issue. Using version 9.7.1 put I'm running php 7.3.
Going to try using `$csv->setEscape('');`

**Comment by mherbold** [comment]
Hello all - I was able to resolve my issue by writing my own version of a CSV exporter instead of using 9.7.1 on PHP 5.6.  Here is the code I wrote, and hopefully this can help others with the same issue.  This code is designed to be a drop-in replacement for league 9.7.1, as long as you are using only the createFromStream and insertOne functions.  It works with multibyte strings as well.  After exporting the CSV data, I then use the latest version of league for PHP 7.4 to import, and it reads in these files perfectly.

```
class Writer
{
	private $resource;
	
	public static function createFromStream( $resource )
	{
		return new Writer( $resource );
	}
	
	public function __construct( $resource )
	{
		$this->resource = $resource;
		
		fputs( $this->resource, chr( 0xEF ) . chr( 0xBB ) . chr( 0xBF ) );
	}
	
	public function insertOne( $data )
	{
		$line = [];
		
		foreach ( $data as $datum )
		{
			if ( mb_ereg( "[,\"\\r\\n]", $datum ) )
			{
				$datum = mb_ereg_replace( '"', '""', $datum );
				
				$line[] = "\"$datum\"";
			}
			else
			{
				$line[] = $datum;
			}
		}
		
		$line = implode( ',', $line );
		
		fputs( $this->resource, "$line\r\n" );
	}
}
```

**Comment by PascalBrouwers** [comment]
Weird things are happening with version 9.7.1 and `$csv->setEscape('');`
Somehow it skips every row that does not have a `\` in it in 1 of the columns.

**Comment by nyamsprod** [comment]
@PascalBrouwers if you can have a simple reproducible case I may investigate it. If you are using PHP7.4+ you will automatically use PHP own fix. For PHP7.2 and PHHP7.3 the package uses a polyfill in userland.

**Comment by PascalBrouwers** [comment]
It seems my csv is incorrect and missing a column for some rows and the polyfill is filtering them out with:
```
if ([null] === $record || !in_array(null, $record, true)) {
```

**Comment by nyamsprod** [comment]
@PascalBrouwers strange indeed if you could isolate the row an some row arounds I would check it as the rules are the same used in the PHP implementation. Would have been nice to compare with a server on PHP7.4

**Comment by PascalBrouwers** [comment]
It's the program that saved the csv that's the culprit. Somehow it saved some rows with 34 commas, others with 35 commas resulting in some rows with `null` values
